pygdf/series.py

```diff
@@ -409,6 +412,40 @@ def dtype(self):
         """dtype of the Series"""
         return self._impl.dtype
 
+    @classmethod
+    def _concat(cls, objs, index=True):
+        head = objs[0]
+        for o in objs:
+            if o._impl != head._impl:
+                raise ValueError("All series must be of same type")
+
+        data = head._impl.concat(objs)
```

Comment:
For categorical data we need to allocate a buffer of `series.cat.codes.dtype`, while for numeric we need to allocate a buffer of `series.dtype`. I figured it made more sense to special case inside the implementations than check if the series was categorical and special case in the method.